docs/api.md

```diff
@@ -0,0 +1,65 @@
+<!-- Generated by documentation.js. Update this documentation by updating the source code. -->
```

Comment:
Should we add this to .gitignore? It's a generated file.

Reply:
I personally liked having the resulting markdown available if someone, didn't want/couldn't use mkdocs to generate it themselves, or couldn't get to the web to see the resulting output.

I don't have a strong opinion either way but having it checked in makes things easier to find and in the HTML generation pipeline...

Reply:
If we have this hooked up to our CI, people can just check out our Github page, which should be sufficient for 99% of people. (For the 1%, they don't actually need MkDocs, since they can just run `npm run doc` and get the Markdown file to look at.)

Mainly, I just don't want this file to get out of sync (and I'm not a huge fan of automatic post-commit changes or things of that nature).

Reply:
Also, at some point I'd like to create a fix for mkdocs/mkdocs#193, which will let us host multiple versions of our docs in one place. I have some ideas for that, but it's not a super-high priority for me at the moment.